### PR TITLE
prosody: visitor mode support

### PR DESCRIPTION
**Pull request by aaronkvanmeerten**

### CODE REVIEW

File: prosody/rootfs/defaults/conf.d/visitors.cfg.lua [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+    ssl = {
+        key = "/config/certs/v{{ $VISITOR_INDEX }}.{{ $XMPP_DOMAIN }}.key";
+        certificate = "/config/certs/v{{ $VISITOR_INDEX }}.{{ $XMPP_DOMAIN }}.crt";
+    }
+    modules_enabled = {
+      'bosh';
+      'ping';
+      "external_services";
+      "smacks";
+      "jiconop";
```
Comment (saghul) [inline prosody/rootfs/defaults/conf.d/visitors.cfg.lua:123]: Do we still need this?

File: prosody/rootfs/defaults/conf.d/visitors.cfg.lua [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+    }
+    main_muc = 'conference.v{{ $VISITOR_INDEX }}.{{ $XMPP_DOMAIN }}';
```
Comment (saghul) [inline prosody/rootfs/defaults/conf.d/visitors.cfg.lua:125]: Why are we hardcoding the MUC name? The base config doesn't.

File: prosody/rootfs/defaults/prosody.cfg.lua [hunk 3]
```diff
@@ -143,6 +165,42 @@ c2s_interfaces = { "*", "::" }
 c2s_interfaces = { "*" }
 {{ end }}
 
+{{ if $ENABLE_S2S -}}
+-- set s2s port
+s2s_ports = { {{ $S2S_PORT }} } -- Listen on specific s2s port
+
+s2s_whitelist = {
+	{{ if $ENABLE_VISITORS -}}
+    'conference.{{ $XMPP_DOMAIN }}'; -- needed for visitors to send messages to main room
```
Comment (saghul) [inline prosody/rootfs/defaults/prosody.cfg.lua:174]: This should probably use the variable or won't work, will it?

File: prosody/rootfs/etc/cont-init.d/10-config [hunk 4]
```diff
@@ -25,10 +25,24 @@ if [[ "$(stat -c %U /prosody-plugins-custom)" != "prosody" ]]; then
...
+    echo "Prosody visitor mode, using alternate config"
+    PROSODY_SITE_CFG="visitors.cfg.lua"
+    rm /config/conf.d/jitsi-meet.cfg.lua
+    # force jicofo into auth domain for visitor-mode prosody
+    [ -z "$XMPP_AUTH_DOMAIN" ] && XMPP_AUTH_DOMAIN="auth.meet.jitsi"
+    export PROSODY_ADMINS="focus@$XMPP_AUTH_DOMAIN"
+else
+    echo "Prosody normal mode, using default config"
+    PROSODY_SITE_CFG="jitsi-meet.cfg.lua"
+    rm /config/conf.d/visitors.cfg.lua
```
Comment (saghul) [inline prosody/rootfs/etc/cont-init.d/10-config:42]: Would it make sense to delete all *.lua from there at the start?

**Review by saghul — COMMENTED** [review]
Left some comments, PTAL.

File: prosody/rootfs/etc/cont-init.d/10-config (hunk 4 above)
Comment (aaronkvanmeerten) [reply prosody/rootfs/etc/cont-init.d/10-config:42]: I was pondering that, but then considered that someone might want to mount other files in there to provide custom configuration, so I didn't want to assume anything there.

File: prosody/rootfs/defaults/conf.d/visitors.cfg.lua (hunk 1 above)
Comment (aaronkvanmeerten) [reply prosody/rootfs/defaults/conf.d/visitors.cfg.lua:123]: Yes, this is the mod_jiconop module, not the deprecated node service unfortunately named the same thing.

File: prosody/rootfs/defaults/conf.d/visitors.cfg.lua (hunk 2 above)
Comment (aaronkvanmeerten) [reply prosody/rootfs/defaults/conf.d/visitors.cfg.lua:125]: Good point, I'll work on this next.

File: prosody/rootfs/defaults/prosody.cfg.lua (hunk 3 above)
Comment (aaronkvanmeerten) [reply prosody/rootfs/defaults/prosody.cfg.lua:174]: Excellent point, good catch.

File: prosody/rootfs/etc/cont-init.d/10-config (hunk 4 above)
Comment (saghul) [reply prosody/rootfs/etc/cont-init.d/10-config:42]: Good point!

**Review by saghul — APPROVED** [review]

**Pull request merged**